src/oop/special_class_method/medium/q10/MainTest.java

```diff
@@ -0,0 +1,217 @@
+                Arguments.of(12, "Blue", 10),
+                Arguments.of(43, "White", 37),
+                Arguments.of(7, "Yellow", 4)
+
+        );
+    }
+
+    private static Stream<Arguments> updateStringInputProvider() {
+        return Stream.of(
+                Arguments.of(3, "Red", "White"),
+                Arguments.of(10, "Black", "Green"),
+                Arguments.of(12, "Blue", "Red"),
+                Arguments.of(43, "White", "Gray"),
+                Arguments.of(7, "Yellow", "Orange")
+
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void correctGetSizeMethod(int value1, String value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, int.class},
+                {value2, String.class}
+        };
+        Object testInstance = testClass.createInstance(arguments);
+        Object getMethodOutput = testClass.callMethod(getAttributeMethodName1, testInstance);
+        assertTrue(testClass.hasModifier(getAttributeMethodName1, null, "public"),
+                "Your " + getAttributeMethodName1 + " method does not have the correct visibility modifier.");
+        _assertEquals(value1, getMethodOutput,
+                "Your " + getAttributeMethodName1 + " method does not return the value of the " + attributeName1 + " attribute.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void correctGetColourMethod(int value1, String value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, int.class},
+                {value2, String.class}
+        };
+        Object testInstance = testClass.createInstance(arguments);
+        Object getMethodOutput = testClass.callMethod(getAttributeMethodName2, testInstance);
+        assertTrue(testClass.hasModifier(getAttributeMethodName2, null, "public"),
+                "Your " + getAttributeMethodName2 + " method does not have the correct visibility modifier.");
+        _assertEquals(value2, getMethodOutput,
+                "Your " + getAttributeMethodName2 + " method does not return the value of the " + attributeName2 + " attribute.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("updateIntegerInputProvider")
+    public void correctSetSizeMethod(int initialiValue, String value2, int updatedValue) throws Throwable {
+        Object[][] instantiationArguments = {
+                {initialiValue, int.class},
+                {value2, String.class}
+        };
+        Object testInstance = testClass.createInstance(instantiationArguments);
+        Object[][] setMethodArguments = {
+                {updatedValue, int.class}
+        };
+        Class<?>[] methodModifierClasses = {
+                int.class
+        };
+        assertTrue(testClass.hasMethod(setAttributeMethodName1, methodModifierClasses, Void.TYPE),
+                "Your " + setAttributeMethodName1 + " method does not have the correct name, return type, or parameters.");
```

Comment:
split the return type into a different assert for message clarity. Do the same for the other set method test